app/src/main/java/de/westnordost/streetcomplete/quests/list_quests/ListQuestForm.kt

```diff
@@ -0,0 +1,61 @@
+package de.westnordost.streetcomplete.quests.list_quests
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.accepts_cards.CardAcceptance.CREDIT_CARDS_ONLY
+import de.westnordost.streetcomplete.quests.accepts_cards.CardAcceptance.DEBIT_AND_CREDIT
+import de.westnordost.streetcomplete.quests.accepts_cards.CardAcceptance.DEBIT_CARDS_ONLY
+import de.westnordost.streetcomplete.quests.accepts_cards.CardAcceptance.NEITHER_DEBIT_NOR_CREDIT
+
+@Composable
+fun <T> ListQuestForm(options: List<TextItem<T>>, onSelectionChange: (TextItem<T>) -> Unit, currentOption: TextItem<T>?, modifier: Modifier = Modifier) {
+    Column(
+        modifier = modifier
+    ) {
+        options.forEach {
+            Row(modifier = Modifier
+                .padding(vertical = 8.dp)
+                .selectable(
+                    selected = (it == currentOption),
```

Comment:
I wonder, is the change animated or is it changed immediately on clicking a different item?

Reply:
![ScreenRecording2025-05-05at1 57 13PM-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/4f1589be-e125-4299-9d57-e5e83e57f14d)

Reply:
ah cool, so that is handled internally in the radio button. I notice a little thing here but will comment in the code